Formula/u/unxip.rb

```diff
@@ -16,15 +16,9 @@ class Unxip < Formula
     sha256 cellar: :any_skip_relocation, monterey:       "039ceba020c663073c09912b6c25230a4735cff3bf775074e5d3e4bba5591c1d"
   end
 
-  depends_on macos: :monterey
-
-  uses_from_macos "swift"
-
-  # Uses Compression framework on macOS
-  on_linux do
-    depends_on "xz"
-    depends_on "zlib"
-  end
+  # requires Swift 6.0
+  depends_on xcode: ["16.0", :build]
+  depends_on :macos
```

Comment:
Are we in a hurry to ship this? Or can we wait for `swift` 6.0?

Reply:
Our macOS 14 runners are still not on Xcode 16 so this should probably wait for that anyway. Was largely waiting for the first PR to need it to give some time to fix any major Xcode 16 build issues so can schedule some maintenance time in the next few days.

Reply:
yeah, I think we can wait for the swift 6.0 build.

Reply:
Definitely should wait as we are losing all Intel bottles given there we don't run Sequoia runner.

Reply:
Not in a hurry, so yes, waiting for `swift` 6.0 seems fine. I've never tried to build this code using the OSS swift instead of Apple's distribution, will be interested how that goes since it still relies on some macOS libs.

Reply:
On side note, I would recommend avoiding `depends_on xcode:` unless there is a hard requirement on XCode.app.

`unxip` can be built with just CLT so we should not force an unnecessary dependency.

Reply:
We would probably want a new DSL because sometimes Swift breaks when using an older SDK because Swift.swiftmodule ships with the SDK.

`depends_on :xcode` also affects SDK selection logic in a way that's beneficial for Swift projects specifically.